provision-osx/tasks/configure_buildfarm_proxy.yml

```diff
@@ -0,0 +1,33 @@
+---
+
+-
+  name: Set proxy vars
+  set_fact:
+    proxy_state: "off"
+  when: proxy_host is undefined or proxy_host == ""
+
+-
+  name: Set node proxy configuration with authentication
+  shell: "networksetup -set{{ item }} {{ proxy_device }} {{ proxy_host }} {{ proxy_port }} on {{ proxy_user }} {{ proxy_pass }}"
+  with_items: "{{ proxy_ctx }}"
+  become: true
+  when: 
+    - proxy_host is defined and proxy_host != ''
+    - proxy_user is defined and proxy_user != ''
+    - proxy_pass is defined and proxy_pass != ''
+
+-
+  name: Set node proxy configuration without authentication
+  shell: "networksetup -set{{ item }} {{ proxy_device }} {{ proxy_host }} {{ proxy_port }} off"
+  with_items: "{{ proxy_ctx }}"
+  become: true
+  when:
+    - proxy_host is defined and proxy_host != ''
+    - proxy_user is not defined or proxy_user == ''
+    - proxy_pass is not defined or proxy_pass == ''
+
+-
+  name: Sets proxy status
+  shell: "networksetup -set{{ item }}state {{ proxy_device }} {{ proxy_state | default('on') }}"
```

Comment:
well kind of... if I add a proxy_hist but run the playbook again with an empty proxy_host I need to run this task otherwise the proxy won't be disabled.